src/AzureExtension/DataModel/DataObjects/Identity.cs

```diff
@@ -181,11 +208,33 @@ public static Identity GetOrCreateIdentity(DataStore dataStore, IdentityRef? ide
         // We don't want to create an identity object and download a new avatar unless it needs to
         // be updated. In the event of an empty avatar we will retry more frequently to update it,
         // but not every time.
-        if (existing is null || ((DateTime.Now.Ticks - existing.TimeUpdated) > _updateThreshold)
+        if (existing is null || (isDeveloper && !existing.Developer) || ((DateTime.Now.Ticks - existing.TimeUpdated) > _updateThreshold)
             || (string.IsNullOrEmpty(existing.Avatar) && ((DateTime.Now.Ticks - existing.TimeUpdated) > _avatarRetryDelay)))
         {
             var newIdentity = CreateFromIdentityRef(identityRef, connection);
-            return AddOrUpdateIdentity(dataStore, newIdentity);
+            return AddOrUpdateIdentity(dataStore, newIdentity, isDeveloper);
+        }
+
+        return existing;
+    }
+
+    // Creation from an Azure Identity object.
+    public static Identity GetOrCreateIdentity(DataStore dataStore, Microsoft.VisualStudio.Services.Identity.Identity? identity, VssConnection connection, bool isDeveloper = false)
```

Comment:
What is `isDeveloper` for?  I see its usage on line 233.  Would a non-developer use DevHome, if they did, why does this information need to be used?

Reply:
Identities are actually created for every person referenced in any pull request or work item, not just the developer. If you create a query widget and the query lists other people, there is an identity record for each person. We don't actually know when a developer is the same identity as a person unless we make that association at the time the identity is added or updated. That's why the property was added, so we can now which identities correspond to the logged in developer. This will be used for notifications and other data in the future, I added it since creating the association when we create the identity makes sense.